[#3] Create init backend tests - health check endpoint

ifirmawan opened the pull request:
## TODO/DONE
- [x] Create init tests
- [x] Update the CI pipelines
- [x] Add a badge for coverage to README.md

### Update the CI pipelines
- [x] Should run automatically on branches with an associated pull requests
- [x] Should run automatically on changes to main
- [x] Should only run when python changes

ifirmawan committed:
[#3] Create init backend tests - health check endpoint

ifirmawan committed:
[#3] Fix CI: Adding alter user createdb

For some reason the backend tests failed due "permission denied to create a database"

ifirmawan committed:
Merge branch 'feature/1-setup-jest' into feature/3-setup-python-testing-with-coveralls

ifirmawan committed:
[#3] Add coverage & coveralls backend deps

ifirmawan committed:
[#3] Create .coveragerc file

ifirmawan committed:
[#3] Update test.sh to run coveralls

ifirmawan committed:
[#3] Add COVERALLS_REPO_TOKEN var in CI

ifirmawan committed:
[#3] Add a badge for coverage to README.md

dedenbangkit commented on backend/requirements.txt:
```diff
@@ -18,3 +18,7 @@ typing_extensions==4.12.2
+
+# code coverage
+coverage==6.2
+coveralls==3.3.1
```
please use latest coverage and coveralls version

ifirmawan replied:
noted, will do

ifirmawan committed:
Merge branch 'feature/1-setup-jest' into feature/3-setup-python-testing-with-coveralls

ifirmawan committed:
[#3] Upgrade coverage & coveralls with latest version

ifirmawan committed:
[#3] Rename secret var

ifirmawan committed:
[#3] Update coveralls badge in readme

ifirmawan committed:
[#3] Remove branch query params in coveralls badge readme

ifirmawan committed:
[#3] Add coveralls var in docker-compose file

ifirmawan committed:
[#3] Run coveralls with service params

Try to fix issue: Failed collecting git data. Are you running coveralls inside a git repository? Is git installed?

ifirmawan committed:
[#3] Revert changes - coveralls in test.sh

ifirmawan committed:
[#3] Add relative_files = True in coveragerc

dedenbangkit committed:
[#3] Add git folder to backend

dedenbangkit committed:
[#3] add docker compose test

dedenbangkit committed:
[#3] Add docker compose test
